source/common/upstream/outlier_detection_impl.cc

```diff
@@ -241,7 +245,8 @@ void DetectorImpl::onConsecutive5xxWorker(HostSharedPtr host) {
 const double Utility::SUCCESS_RATE_STDEV_FACTOR = 1.9;
 
 double Utility::successRateEjectionThreshold(
-    double success_rate_sum, const std::vector<HostSuccessRatePair>& valid_success_rate_hosts) {
+    double success_rate_sum, const std::vector<HostSuccessRatePair>& valid_success_rate_hosts,
+    double& success_rate_average) {
```

Comment:
Instead of passing an out param, can you return a struct. It's easier to grok what is going on.